### ISSUE TITLE
Types of parameters 'descriptor' and 'descriptor' are incompatible.

### ISSUE THREAD

**Issue by Zoddo** [Papooch/decorate-all#7]
Hello,

When trying to use `@DecorateAll(ApiParam({/* ... */]))` (from `@nestjs/swagger`), TypeScript complains with the following error :
```
Argument of type 'MethodDecorator' is not assignable to parameter of type '(target: any, propertyKey: string, descriptor?: PropertyDescriptor | undefined) => void'.
  Types of parameters 'descriptor' and 'descriptor' are incompatible.
    Type 'PropertyDescriptor | undefined' is not assignable to type 'TypedPropertyDescriptor<any>'.
      Type 'undefined' is not assignable to type 'TypedPropertyDescriptor<any>'.
```

Shouldn't the `decorator` argument (of `DecorateAll()`) below simply be of type `MethodDecorator`?
https://github.com/Papooch/decorate-all/blob/a364469cb6ea814bddbc001b4d3403b560ca01cf/src/lib/decorate-all.decorator.ts#L17-L21

**Comment by Papooch** [comment]
`ApiParam` is a property decorator.

Are you trying to decorate properties that are not methods? That is unfortunately not possible and it's a limitation of typescript. It's why NestJS has to use swagger-cli plugin as an intermediate compilation step.

**Comment by Zoddo** [comment]
> `ApiParam` is a property decorator.

`ApiParam` is a **method** decorator (you use inside controllers to describe parameters inside routes like `/users/:id`). You even used it here as an example: https://discord.com/channels/520622812742811698/527863537708695562/906493595568185344 (from NestJS's Discord server; that's how I discovered this very useful package btw :smiley:)

However, `ApiProperty` is a property decorator :wink:

**Comment by Papooch** [comment]
Oh, yeah, I had `ApiProperty` in mind, sorry.! I'll look into it closer and make the types compatible.

**Comment by Papooch** [comment]
Fixed in `1.2.1`
